File: src/main/java/eterea/migration/api/rest/model/Payment.java
package eterea.migration.api.rest.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.OffsetDateTime;
import java.util.List;

@EqualsAndHashCode(callSuper = true)
@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table
public class Payment extends Auditable {

    @Id
    private Long orderNumberId;
    private String transaccionComercioId;
    private String transaccionPlataformaId;
    private String tipo;
    private BigDecimal monto;
    private String estado;

    @Lob
    private String detalle;
    private String metodoPago;
    private String medioPago;
    private Integer estadoId;
    private Integer cuotas;
    private String informacionAdicional;
    private String marcaTarjeta;
    private String informacionAdicionalLink;
    private OffsetDateTime fechaTransaccion;
    private OffsetDateTime fechaPago;

    @OneToOne
    @JoinColumn(name = "orderNumberId", insertable = false, updatable = false)
    private InformacionPagador informacionPagador;

    @OneToMany(fetch = FetchType.EAGER)
    @JoinColumn(name = "orderNumberId", insertable = false, updatable = false)
    private List<ProductTransaction> productTransactions;
}

File: src/main/kotlin/eterea/migration/api/rest/kotlin/model/Payment.kt
package eterea.migration.api.rest.kotlin.model

import jakarta.persistence.*
import java.math.BigDecimal
import java.time.OffsetDateTime

@Entity
@Table
data class Payment(

    @Id
    var orderNumberId: Long? = null,
    var transaccionComercioId: String = "",
    var transaccionPlataformaId: String = "",
    var tipo: String = "",
    var monto: BigDecimal = BigDecimal.ZERO,
    var estado: String = "",

    @Column(columnDefinition = "TEXT CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci")
    var detalle: String = "",
    var metodoPago: String? = null,
    var medioPago: String? = null,
    var estadoId: Int? = null,
    var cuotas: Int? = null,
    var informacionAdicional: String? = null,
    var marcaTarjeta: String? = null,
    var informacionAdicionalLink: String? = null,
    var fechaTransaccion: OffsetDateTime? = null,
    var fechaPago: OffsetDateTime? = null,

    @OneToOne
    @JoinColumn(name = "orderNumberId", insertable = false, updatable = false)
    var informacionPagador: InformacionPagador? = null,

    @OneToMany
    @JoinColumn(name = "orderNumberId", insertable = false, updatable = false)
    var productTransactions: List<ProductTransaction?>? = null

) : Auditable()
